src/library.js

```diff
@@ -784,7 +784,15 @@ LibraryManager.library = {
     abort('Assertion failed: ' + (condition ? UTF8ToString(condition) : 'unknown condition') + ', at: ' + [filename ? UTF8ToString(filename) : 'unknown filename', line, func ? UTF8ToString(func) : 'unknown function']);
   },
 
-  terminate__sig: 'vi',
+  __cxa_call_unexpected__sig: 'vi',
+  __cxa_call_unexpected: function(exception) {
+    err('Unexpected exception thrown, this is not properly supported - aborting');
+#if !MINIMAL_RUNTIME
+    ABORT = true;
+#endif
+    throw exception;
+  },
+
   terminate: '__cxa_call_unexpected',
```

Comment:
Oh I see... shouldn't __cxa_call_unexpected stay in the other file and be be implemented in terms of `terminate` rather than the other way around?

Reply:
How about this:

```
__cxa_call_unexpected__deps: ['terminate']
__cxa_call_unexpected: function(exception) {
  err('Unexpected exception thrown (' + exception + '), this is not properly supported - aborting');
  terminate()
}
```

```
  terminate__sig: 'v',
￼  terminate: function() {
￼#if !MINIMAL_RUNTIME
￼    ABORT = true;
￼#endif
￼    throw "terminated";
￼  },
```

Reply:
Alternative do you even need terminate here?  Can we just remove it?

Reply:
Well, we have shared `__cxa` stuff in `library.js` already. This function doesn't diverge in exceptions vs stub, so we may as well have it in the single shared place, that's shorter and cleaner I think?

Not sure why we should reverse the order of what's implemented in terms of what..? I'm not following your logic there.

Reply:
I guess my concern is.. why is "terminate" and alias for `__cxa_call_unexpected`?     Is far as I can tell terminate takes zero args so they should no be aliases.  Also if any actually calls terminate they should not see "Unexpected exception thrown" should they?

Reply:
I guess that was an existing error, fair point. Fixed to implement terminate separately.